File: 1756-minimum-deletions-to-make-string-balanced/1756-minimum-deletions-to-make-string-balanced.java
class Solution {
    public int minimumDeletions(String s) {
        int min_del = 0;
        int b_cnt = 0;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == 'b') b_cnt++;
            else min_del = Math.min(min_del + 1, b_cnt);
        }

        return min_del;
    }
}

File: 1756-minimum-deletions-to-make-string-balanced/1756-minimum-deletions-to-make-string-balanced.kt
class Solution {
    fun minimumDeletions(s: String): Int {
        var a_cnt : Int = 0;
        for (c in s) if (c == 'a') a_cnt++;

        var b_cnt : Int = 0;
        var min_del = s.length;

        for (c in s) {
            if (c == 'a') a_cnt--;
            min_del = Math.min(min_del, a_cnt + b_cnt);
            if (c == 'b') b_cnt++;
        }

        return min_del;
    }
}